soc: arm: stm32wb serie low power modes substates

FRASTM opened the pull request:
Change the substate-id range from 1-3 as described in the include/power/power_state.h
this is changing the range in pm_power_state_set() of the soc/arm/st_stm32/stm32wb/power.c
The debug config will let the clocks active in STOP mode at init.
Following the https://github.com/zephyrproject-rtos/zephyr/pull/33131, the power.c part of the stm32wb is modified in the same way.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on soc/arm/st_stm32/stm32wb/Kconfig.defconfig.series:
```diff
@@ -12,11 +12,9 @@ config SOC_SERIES
 
 if PM
 config PM_DEVICE
-	bool
 	default y
 
 config STM32_LPTIM_TIMER
-	bool
```
Nitpick: A "+ minor fixes" in the commit message for this would have been nice

FRASTM replied:
added in the commit msg

FRASTM committed:
soc: arm: stm32wb serie low power modes in debug config

The debug config will let the clocks active in STOP mode
at init.
The substate-id is mapping the same Zephyr power state.
Plus a minor fix in the Kconfig.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
soc: arm: stm32wb low power substates

The low power states 0∕1∕2 are added to the stm32wb.
The substate-id is mapping the same Zephyr power state.
They correspond to the stm32wb low power stop0/1/2 modes.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
boards: arm: stm32wb55 nucleo board has low power states

The low power states 0∕1∕2 are added to the stm32wb55rg
nucleo board, with default min residency values.
The substate-id is mapping the same Zephyr power state.
They correspond to the stm32wb55 low power stop0/1/2 modes.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
rebase and change the substate range

ABOSTM reviewed (changes requested):
Commit Message:
> . . .
> The low power states 0∕1∕2 are added to the stm32wb55rg
> nucleo board, with (default min residency values).

Sentence not finished: With what ? (or wrong parenthesis)

Otherwise LGTM

FRASTM commented:
rebase
changed info.substate_id in the pm_power_state_exit_post_ops function

ABOSTM reviewed (approved):
LGTM
